### ISSUE TITLE
ToS compliance: is this legit like GW2Launcher?

### ISSUE THREAD

**Issue by lukycody** [Naylepsh/guild-wars-2-auto-daily-rewards#4]
Hey there. Nice little python script to automate the daily login. I like it, as its quite simple too!

One question though: is using this against the ToS/GW2 policy, or is it okay?
I'm not sure - it's not doing more than one action/press per Person and account at any given time (multibox farming or smth).
If you delete the step "open_reward()" it doesn't even do any actual actions in the game with your character.
As such it appears to fall into the same category as e.g. [GW2Launcher](https://github.com/Healix/Gw2Launcher), which also can automatically login and startup several account one after another. The only difference would be selecting the character to login with.

However they Anets lists this scenario as being not allowed:
"You cannot use macros for the sole purpose of farming items, currencies, karma, or other benefits in the game, particularly while away from the keyboard (AFK)"
https://help.guildwars2.com/hc/en-us/articles/360013762153-Policy-Macros-and-Macro-Use

**Comment by Naylepsh** [comment]
Hey. Thanks for showing an interest in the script :)

To be honest, I have no clue about the ToS compliance of the script -- I have not reached out to the ArenaNet and frankly I doubt that I would had gotten a response anyway due to the way they handle (or handled, don't follow gw2 much nowadays) such 3rd party tools, which is: say nothing until it's really pushing the boundaries (things like arcdps predecessor come to mind...)

This is quite anecdotic, but historically what ArenaNet meant by farming while AFK was something akin to botting while you're away for an extended period of time. Things such as scripts clicking Reaper's GS #4 or automated Engi turrets for farming karma / loot bags were okay-ish as long as you could respond to an admin's message on chat within a reasonable timeframe. For more info on these 2 cases I recommend searching on gw2 subreddit.

As for the detecting part -- They theorically could measure stuff like timing between inputs or input placements (as these aren't random in this script's case) and they could act on suspicious patterns. But such things could also detect innocent scripts like macros for instruments.


TLDR: I have no clue, you should use it at your own risk, but it has such a minimal impact on the game, that realistically they'd have to ban a large portion of community if they were to punish for stuff so insignificant as this.

**Comment by lukycody** [comment]
Thanks for the reply!
Well I guess theres nothing wrong with using it to complete daily login _while_ sitting at the PC/desk doing study-stuff.
You just can't do smth else with the PC (browsing etc.) while the script runs, as it will break the focus of the programm, right? (so letting it alone as it completes)

**Comment by Naylepsh** [comment]
Yep, that's correct. GW2 has to be the active window for the script to work.

**Comment by ChoriSw** [comment]
> Thanks for the reply! Well, I guess then theres nothing wrong with using it _while_ sitting at the PC/desk doing desk-work or study-stuff. I suppose you just can't do smth else with the PC (browsing etc.) while the script runs, as it will break the focus of the programm, right? (so you need to leave the inputs alone as the script runs through and completes)

Heya did u try it , how's it going so far?

**Comment by lukycody** [comment]
Hey! Sorry, no haven‘t tried yet. Why are you asking? Aren‘t you guys developing it using it yourself?

Do you mean usability-wise or legit-use-wise though?
